src/Http/Http.Extensions/src/JsonOptions.cs

```diff
@@ -26,7 +27,7 @@ public class JsonOptions
         // The JsonSerializerOptions.GetTypeInfo method is called directly and needs a defined resolver
         // setting the default resolver (reflection-based) but the user can overwrite it directly or calling
         // .AddContext<TContext>()
-        TypeInfoResolver = CreateDefaultTypeResolver()
+        TypeInfoResolver = TrimmingAppContextSwitches.EnsureJsonTrimmability ? null : CreateDefaultTypeResolver()
     };
 
     // Use a copy so the defaults are not modified.
```

Comment:
What's the plan for the suppressions on line 39-44?

Reply:
For now, they will suppress in the `Microsoft.AspNetCore.Http.Extensions.WarningSuppressions.xml` and I will file an issue to start a discussion if we need something similar to runtime (`LibraryBuild.xml`).